dsindex.go

```diff
@@ -51,7 +56,68 @@ func (ei entityIndex) String() string {
 
 type DatastoreIndex map[string][]entityIndex
 
-func LoadIndex(data []byte) (DatastoreIndex, error) {
+type datastoreAdminClient struct {
+	adapter datastoreAdminAdapter
+}
+
+type datastoreAdminAdapter interface {
+	withEachIndexFrom(*dsadmin.ListIndexesRequest, func(index *dsadmin.Index)) error
+}
+
+type datastoreAdminAdapterImpl struct{
+	client *admin.DatastoreAdminClient
+	ctx context.Context
+}
+
+func (d datastoreAdminAdapterImpl) withEachIndexFrom(request *dsadmin.ListIndexesRequest, f func(index *dsadmin.Index)) error {
+	it := d.client.ListIndexes(d.ctx, request)
+	for {
+		item, err := it.Next()
+		if err == iterator.Done {
+			break
+		} else if err != nil {
+			return fmt.Errorf("error listing indexes: %s", err.Error())
+		}
+		f(item)
+	}
+	return nil
+}
+
+func NewDatastoreAdminClient(ctx context.Context) datastoreAdminClient {
+	c, err := admin.NewDatastoreAdminClient(ctx)
+	if err != nil {
+		panic(fmt.Sprintf("Error creating DatastoreAdminClient: %s", err))
+	}
+
+	dac := datastoreAdminClient{
+		adapter:  &datastoreAdminAdapterImpl{client: c, ctx: ctx},
+	}
+	return dac
+}
+
+func (c datastoreAdminClient) GetDatastoreIndex(project string) (DatastoreIndex, error) {
+	req := &dsadmin.ListIndexesRequest{
+		ProjectId: project,
+		Filter: "state=READY",
```

Comment:
Thinking we don't need all indexes anymore? - Regarding the removal of onlyReady (I think it was called)